vm-device/src/interrupt/mod.rs

```diff
@@ -68,6 +68,11 @@ pub type Result<T> = std::io::Result<T>;
 /// Data type to store an interrupt source identifier.
 pub type InterruptIndex = u32;
 
+pub const IRQ_KEEP_MASKED: usize = 1 << 0;
+pub const IRQ_KEEP_UNMASKED: usize = 1 << 1;
+pub const IRQ_MASKED_TO_UNMASKED: usize = 1 << 2;
+pub const IRQ_UNMASKED_TO_MASKED: usize = 1 << 3;
+
```

Comment:
This could be an `enum` but it seems a bit of weird way to implement this - why not store the previous mask value and check if the mask request has changed and then update?

Reply:
Yes, using enum here makes the code look neater, but we need to implement BitOr and PartialEq, which increases the complexity of the code and I think may be more error-prone.
Storing the previous mask value is a good idea, however the mask state is a state of gsi, and we need to store a HashMap or something like that in InterruptSourceGroup, so I suppose the code in the PR is the more simple way.
Could we keep this usize code plz? :)